### PR TITLE
IDAM-83 Callback

### PR DESCRIPTION
**Pull request by AlzAmido**

### CODE REVIEW

File: custom-pages/login/src/services/AuthApi.js [hunk 1]
```diff
@@ -29,22 +29,22 @@ export default class AuthApi {
...
         realm: authOpts.connection,
         username,
         password
       },
       (err) => {
         if (err) {
-          if (!err.message) {
-            err.message = 'Invalid username or password'
+          if (cb) {
+            setTimeout(() => cb('Invalid username or password'), 5)
```
Comment (John-Williams) [inline custom-pages/login/src/services/AuthApi.js]: why the setTimeout?

File: custom-pages/login/src/services/AuthApi.js [hunk 2]
```diff
@@ -29,22 +29,22 @@ export default class AuthApi {
     this.instance = new auth0.WebAuth(params)
   }
 
-  async login(username, password) {
-    return this.instance.login(
+  login(username, password, cb) {
```
Comment (John-Williams) [inline custom-pages/login/src/services/AuthApi.js]: can cb be "callback" please? just so it's clear.

File: custom-pages/login/src/services/AuthApi.js (hunk 1 above)
Comment (AlzAmido) [reply custom-pages/login/src/services/AuthApi.js]: The callback is executed in a service outside of React but it needs triggering a React re-render (in the caller component) so I am forcing the (React) reconciler to go through another cycle using setTimeout

File: custom-pages/login/src/services/AuthApi.js (hunk 2 above)
Comment (AlzAmido) [reply custom-pages/login/src/services/AuthApi.js]: sure, do you mind if I do in the other PR?

Comment (John-Williams) [reply custom-pages/login/src/services/AuthApi.js]: Sure

**Pull request merged**